### PR TITLE
[AWS] [ECS] start working on generators

### PR DESCRIPTION
**Pull request by nagypeterjob**
- [x] create-capacity-provider
- [x] create-cluster
- [x] create-service
- [x] create-task-set
- [x] delete-account-setting
- [x] delete-attributes
- [x] delete-capacity-provider
- [x] delete-cluster
- [x] delete-service
- [x] delete-task-set
- [x] deploy
- [x] deregister-container-instance
- [x] deregister-task-definition
- [x] describe-capacity-providers
- [x] describe-clusters
- [x] describe-container-instances
- [x] describe-services
- [x] describe-task-definition
- [x] describe-task-sets
- [x] describe-tasks
- [x] discover-poll-endpoint
- [x] execute-command
- [x] list-account-settings
- [x] list-attributes
- [x] list-clusters
- [x] list-container-instances
- [x] list-services
- [x] list-tags-for-resource
- [x] list-task-definition-families
- [x] list-task-definitions
- [x] list-tasks
- [x] put-account-setting
- [x] put-account-setting-default
- [x] put-attributes
- [x] put-cluster-capacity-providers
- [x] register-container-instance
- [x] register-task-definition
- [x] run-task
- [x] start-task
- [x] stop-task
- [x] submit-attachment-state-changes
- [x] submit-container-state-change
- [x] submit-task-state-change
- [x] tag-resource
- [x] untag-resource
- [x] update-capacity-provider
- [x] update-cluster
- [x] update-cluster-settings
- [x] update-container-agent
- [x] update-container-instances-state
- [x] update-service
- [x] update-service-primary-task-set
- [x] update-task-set
- [x] wait

### CODE REVIEW

File: dev/aws/ecs.ts [hunk 1]
```diff
@@ -1,3 +1,230 @@
...
+  command: string,
+  options: string[],
+  parentKey: string,
+  childKey = ""
+): Promise<Fig.Suggestion[]> => {
+  try {
+    let cmd = `aws ecs ${command}`;
+
+    for (let i = 0; i < options.length; i++) {
+      const option = options[i];
```
Comment (fedeci) [inline dev/aws/ecs.ts:44]: nit:
```suggestion
    for (const option of options) {
```

File: dev/aws/ecs.ts [hunk 2]
```diff
@@ -1,3 +1,230 @@
...
+      ...dots_arr.sort(sortFnStrings),
+    ];
+  };
+
+  const temp_array = alphabeticalSortFilesAndFolders(arr);
+
+  const final_array = [];
+
+  temp_array.forEach((item) => {
+    if (item !== "" && item !== null) {
```
Comment (fedeci) [inline dev/aws/ecs.ts:172]:
```suggestion
    if (item) {
```

File: dev/aws/ecs.ts [hunk 3]
```diff
@@ -1,3 +1,230 @@
...
+const sortSuggestions = (arr: string[], isS3?: boolean): Fig.Suggestion[] => {
+  const sortFnStrings = (a, b) => {
+    return a.localeCompare(b);
+  };
+
+  const alphabeticalSortFilesAndFolders = (arr) => {
+    const dots_arr = [];
+    const other_arr = [];
+
+    arr.map((fsObject) => {
```
Comment (fedeci) [inline dev/aws/ecs.ts:144]:
```suggestion
    arr.forEach((fsObject) => {
```

File: dev/aws/ecs.ts [hunk 4]
```diff
@@ -1,3 +1,230 @@
...
+    console.log(e);
+  }
+  return [];
+};
+
+const _prefixFile = "file://";
+
+const appendFolderPath = (tokens: string[], prefix: string): string => {
+  const baseLSCommand = "\\ls -1ApL ";
+  let whatHasUserTyped = tokens[tokens.length - 1];
```
Comment (fedeci) [inline dev/aws/ecs.ts:83]: nit: I would name this variable `userInput`

**Review by fedeci — COMMENTED** [review]
I just gave a quick look, will review more deeply tomorrow.

**Comment by QuiiBz** [comment]
Hey @fedeci, thanks for helping with reviews. However, you shouldn't review Draft PRs as they may change in the future.

File: dev/aws/ecs.ts (hunk 4 above)
Comment (nagypeterjob) [reply dev/aws/ecs.ts:136]: Thank you for your suggestions above, but I won't touch them yet, as this code is present in 6-8 other specs. I will fix them up once Fig will support sharing code between specs

**Review by QuiiBz — APPROVED** [review]
LGTM! We will soon be able to share code across specs, so that's gonna be so much easier and cleaner for those aws specs!

**Pull request merged**